ibc/store/proofs_ics23_test.go

```diff
@@ -0,0 +1,180 @@
+package store
+
+import (
+	"crypto/sha256"
+	"testing"
+
+	ics23 "github.com/cosmos/ics23/go"
+	"github.com/pokt-network/pocket/persistence/kvstore"
+	"github.com/pokt-network/smt"
+	"github.com/stretchr/testify/require"
+)
+
+// Proof generation cannot fail but verification can
+func TestICS23Proofs_GenerateCommitmentProofs(t *testing.T) {
+	nodeStore := kvstore.NewMemKVStore()
+	tree := smt.NewSparseMerkleTree(nodeStore, sha256.New(), smt.WithValueHasher(nil))
+	require.NotNil(t, tree)
+
+	// Set a value in the store
+	err := tree.Update([]byte("foo"), []byte("bar"))
+	require.NoError(t, err)
+	err = tree.Update([]byte("bar"), []byte("foo"))
+	require.NoError(t, err)
+	err = tree.Update([]byte("testKey"), []byte("testValue"))
+	require.NoError(t, err)
+	err = tree.Update([]byte("testKey2"), []byte("testValue2"))
+	require.NoError(t, err)
+
+	testCases := []struct {
+		name            string
+		key             []byte
+		value           []byte
+		isNonMembership bool
+		fails           bool
+		expected        error
+	}{
+		{
+			name:            "Successfully generates a membership proof for a key stored",
+			key:             []byte("foo"),
+			value:           []byte("bar"),
+			isNonMembership: false,
+			fails:           false,
+			expected:        nil,
+		},
+		{
+			name:            "Successfully generates a non-membership proof for a key not stored",
+			key:             []byte("baz"),
+			value:           []byte("testValue2"), // unrelated leaf data
+			isNonMembership: true,
+			fails:           false,
+			expected:        nil,
+		},
+		{
+			name:            "Successfully generates a non-membership proof for an unset nil key",
+			key:             nil,
+			value:           []byte("foo"), // unrelated leaf data
+			isNonMembership: true,
+			fails:           false,
+			expected:        nil,
+		},
+	}
+
+	for _, tc := range testCases {
+		t.Run(tc.name, func(t *testing.T) {
+			var proof *ics23.CommitmentProof
+			if tc.isNonMembership {
+				proof, err = createNonMembershipProof(tree, tc.key)
+			} else {
+				proof, err = createMembershipProof(tree, tc.key, tc.value)
+			}
+			if tc.fails {
+				require.EqualError(t, err, tc.expected.Error())
+				require.Nil(t, proof)
+				return
```

Comment:
Noticed this was resolved without a comment. 

I'm okay with the pushback but would like to understand why.


![Screenshot 2023-06-28 at 3 22 07 PM](https://github.com/pokt-network/pocket/assets/1892194/32a490b5-b8f8-4b64-84ca-9ff9f14c1b08)

Reply:
My bad this was actually an error from reverting out of the `t.Run()` way of things with the name. Now its back the return should be there.

Reply:
Got it. Makes more sense now.